rebuild.sh: Add patch to fix Time-Local tests for year 2020

zakame opened the pull request:
- https://rt.cpan.org/Public/Bug/Display.html?id=124787
- https://github.com/Perl/perl5/issues/17410

Commit:
rebuild.sh: Add patch to fix Time-Local tests for year 2020

- https://rt.cpan.org/Public/Bug/Display.html?id=124787
- https://github.com/Perl/perl5/issues/17410

zakame commented on lib/5.26.2/x86_64-linux/CORE/config.h:
```diff
@@ -9,9 +9,9 @@
- * Target system     : linux default-debian-7 3.2.0-5-amd64 #1 smp debian 3.2.96-3 x86_64 gnulinux 
+ * Configuration time: Mon Feb  3 05:15:30 UTC 2020
+ * Configured by     : vagrant
+ * Target system     : linux debian-9 4.9.0-11-amd64 #1 smp debian 4.9.189-3+deb9u2 (2019-11-11) x86_64 gnulinux 
```
Due rebuilding this on an actual machine rather than in a Vagrant VM.

turnstep commented on rebuild.sh:
```diff
@@ -5,6 +5,9 @@ wget http://www.cpan.org/src/5.0/perl-5.26.2.tar.gz
 cd perl-5.26.2
 
+# Apply patch for fixing Time::Local tests entering year 2020  (CPAN RT#124787)
+curl -fsSL https://rt.cpan.org/Public/Ticket/Attachment/1776857/956088/0001-Fix-Time-Local-tests.patch | patch -p1
```
This card is so large already, why don't we just store a copy of the patch itself?

zakame replied:
Yep thinking about that as well, probably better to keep it and maybe later patches in a directory to `cat | patch` through...

zakame replied:
Now added as a local copy to `zcat | patch || exit 1` with.

turnstep commented on bin/perlbug:
```diff
@@ -2,7 +2,7 @@
 	if $running_under_some_shell;
 
-my $config_tag1 = '5.26.2 - Mon Jul 30 11:50:48 UTC 2018';
+my $config_tag1 = '5.26.2 - Mon Feb  3 08:21:42 UTC 2020';
```
I've no idea how this all works, but why is the version and patchlevel_date staying the same, but the date changes?

zakame replied:
This is filled in by the `make`s during build/install, presumably to let one know how recent this perl was configured.

turnstep commented on lib/5.26.2/x86_64-linux/Config_heavy.pl:
```diff
@@ -1038,15 +1038,15 @@ sub myconfig {
 installhtml3dir=''
-installman1dir='/home/git/binary-com/perl/man/man1'
-installman3dir='/home/git/binary-com/perl/man/man3'
+installman1dir=''
```
No man pages?

zakame replied:
Yep, we now have `-Dman1dir=none -Dman3dir=none` in our `./Configure` call.

turnstep commented on lib/5.26.2/x86_64-linux/Config_heavy.pl:
```diff
@@ -1122,12 +1122,12 @@ sub myconfig {
-man3dir='/home/git/binary-com/perl/man/man3'
-man3direxp='/home/git/binary-com/perl/man/man3'
-man3ext='3perl'
+man1dir=' '
```
s/{SPACE}// ?

zakame replied:
Again another automatic `./Configure`/`make` change; we're due to update this anyway in a subsequent clean rebuild over a container.

zakame committed:
Add Time-Local patch directly into new directory

Supply a full compressed copy of the Time-Local patch here so we don't
need to fetch it all the time.

zakame committed:
rebuild.sh: Use local copy of Time-Local patch

Make our patching verbose as well to see the actual files changed.

zakame commented on lib/5.26.2/pod/perltoc.pod:
```diff
@@ -20144,146 +20144,6 @@ utf8, encode, compress, int32, null
 
 =back
 
-=head2 DB_File - Perl5 access to Berkeley DB version 1.x
```
Missed installing Berkeley DB (and GDBM) during the rebuild, these need to be here...

zakame committed:
:gear: rebuild
